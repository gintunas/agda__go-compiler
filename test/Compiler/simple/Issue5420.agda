open import Agda.Builtin.IO
open import Agda.Builtin.Reflection
open import Agda.Builtin.String
open import Agda.Builtin.Unit

_>>=_ : {A B : Set} → TC A → (A → TC B) → TC B
_>>=_ = λ x f → bindTC x f

postulate
  putStr : String → IO ⊤

{-# FOREIGN GHC import qualified Data.Text.IO #-}
{-# COMPILE GHC putStr = Data.Text.IO.putStr #-}
{-# COMPILE JS putStr =
    function (x) {
      return function(cb) {
        process.stdout.write(x); cb(0); }; } #-}

main : IO ⊤
main = putStr "Success\n"
