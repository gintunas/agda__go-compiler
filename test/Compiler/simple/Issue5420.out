EXECUTED_PROGRAM

ret > ExitSuccess
out > Success
out >
